src/Notion.ts

```diff
@@ -495,6 +498,39 @@ export class Notion {
     });
   }
 
+  /**
+ * Activates haptic motors
+ * 
+ * ```typescript
+ * notion.haptics({P7: ["tripleClick100"], P8: ["tripleClick100"]});
+ * ```
+ *
+ * @param effects Effects to produce. Must include both P7 and P8 keys in object. Each key can be an array of up to 7 commands. 
+ *  A list of commands can be found on ./utils/hapticCodes.ts - there are about 127 of them!
+ */
+  public haptics(effects: HapticEffects): any {
+    if (!this.api.didSelectDevice()) {
+      return Promise.reject(errors.mustSelectDevice);
+    }
+
+    if (!effects['P7'] || !effects['P8']) {
+      throw new Error("Notion: an object with keys P7 and P8 required");
+    }
+
+    const maxItems = 7;
+    if (effects['P7'].length > maxItems || effects['P8'].length > maxItems) {
```

Comment:
I would iterate every key in the array and check for the length in every value. This way, the error could contain the name of the motor. E.g:

The maximum effects in ${motorName} is ${maxItems}